File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    total_cnt = 0
    ponket_type = dict()
    for n in nums :
        if n not in ponket_type :
            ponket_type[n] = True
            total_cnt += 1
            
    if total_cnt >= len(nums) // 2 :
        return len(nums)//2
    else :
        return total_cnt

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums)
{
    int type_cnt = 0;
    int N = nums.size();
    vector<int> lst;
    
    for (int n : nums) {
        if (find(lst.begin(), lst.end(), n) == lst.end()) {
            lst.push_back(n);
        }
    }
    
    type_cnt = lst.size();
    
    if (type_cnt >= N/2) {
        return N/2;
    } else {
        return type_cnt;
    }
}